LiverVolumetry/LiverVolumetry.py

```diff
@@ -387,8 +387,14 @@ def computeVolume(self, segmentsVolumeNode, targetSegmentVolumeNode, segmentatio
         segStatLogic.computeStatistics()
         stats = segStatLogic.getStatistics()
         for segmentId in stats["SegmentIDs"]:
-          voxel_count = stats[segmentId,"LabelmapSegmentStatisticsPlugin.voxel_count"]
-          volume_cm3 = stats[segmentId,"LabelmapSegmentStatisticsPlugin.volume_cm3"]
+          voxel_count = 0
+          volume_cm3 = 0
+          if stats[segmentId,"LabelmapSegmentStatisticsPlugin.voxel_count"]:
+            voxel_count = stats[segmentId,"LabelmapSegmentStatisticsPlugin.voxel_count"]
+            volume_cm3 = stats[segmentId,"LabelmapSegmentStatisticsPlugin.volume_cm3"]
+          elif stats[segmentId,"ScalarVolumeSegmentStatisticsPlugin.voxel_count"]:
+            voxel_count = stats[segmentId,"ScalarVolumeSegmentStatisticsPlugin.voxel_count"]
+            volume_cm3 = stats[segmentId,"ScalarVolumeSegmentStatisticsPlugin.volume_cm3"]
```

Comment:
Please explain this snippet. Why the need to have LabelMap and ScalarVolume counts?

Reply:
Although I added the selected segmentations combined node as a "vtkMRMLLabelMapVolumeNode", in some cases, it happens that the input turns into ScalarVolume data and pops up error messages about couldn't find LabelmapSegmentStatisticsPlugin. But lately, I haven't met this issue, just in case it would happen to a different dataset, I added those scripts to ensure that the function works.

Reply:
Region growing setting changes because we have more than 7 labels in the original data, to avoid conflicts, adjust the base value to 100.

Reply:
Resolution setting change is a temporary solution before I develop a stable method. For most cases, it needs at least 500*500 to ensure successful region growing for FaceConnectivity.